circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/event/LoggingListener.java

```diff
@@ -51,33 +51,67 @@
   private ReplicationState replicationState = new ReplicationState();
 
   @VisibleForTesting
-  class ReplicationState {
-    List<String> partitionKeys;
-    int partitionsAltered;
-    long bytesReplicated;
+  static class ReplicationState {
```

Comment:
I think I prefer left side. I this again a complaint that the inner class doesn't have a reference to the outer class? There were loads of references to the inner class though.

Reply:
No, this seems to be related to enforcing encapsulation and it comes from Checkstyle: http://checkstyle.sourceforge.net/apidocs/com/puppycrawl/tools/checkstyle/checks/design/VisibilityModifierCheck.html

Reply:
But I do think that a shorter version is more readable, however there's a warning for each variable that doesn't have a setter and a getter.

Reply:
hmm from that link it should complain because of the annotation?

> ignoreAnnotationCanonicalNames - the list of annotations canonical names which ignore variables in consideration, if user will provide short annotation name that type will match to any named the same type without consideration of package, list by default:
> 
> org.junit.Rule
> org.junit.ClassRule
> com.google.common.annotations.VisibleForTesting

the whole point was to not have a getter/setter to make the code a bit more concise :)

Reply:
Ah, yes, then maybe checkstyle could be updated to not warn us about things with `@VisibleForTesting`?

Reply:
Well, I think generally we would want it to check methods annotated with `@VisibleForTesting` as often these are just "normal" methods/classes that we want to test directly. @patduin I don't think it's complaining because of the annotation it's because the class is visible outside of this class and it exposes its state so a caller from outside can fiddle with its internals which is frowned upon. I'd be OK with extracting this out into its own class and then leaving with the getters/setters.

Reply:
They way I read it was it should not complain because of the annotation.
When would use the annotation on a private member? But reading the  link again it says variables so perhaps methods don't apply. Dunno. if it is private I guess checkstyle won't complain but then if becomes harder to test.
How about we make the ReplicationState class private and expose the fields in the listener like so:
```
  List<String> getPartitionKeys() {
    return Collections.unmodifiableList(replicationState.partitionKeys);
  }
```
Then we can test the state without exposing the class and we don't need a million lines extra.

Reply:
Having a method that exposes the fields sounds good, but it seems that the parameters are the same in the test, before and after
`listener.tableReplicationStart(tableReplication, "event-id");`